### PR TITLE
Fix cluster ID in app pending update rules

### PR DESCRIPTION
**Pull request by rossf7**
This PR:

- Fixes cluster ID in app pending update rules.

### Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: helm/prometheus-rules/templates/alerting-rules/app.rules.yml [hunk 1]
```diff
@@ -107,12 +107,26 @@ spec:
...
+        severity: notify
+        team: batman
+        topic: releng
+    ## Workload cluster app rules only available on the management cluster Prometheus because the source is app-exporter
     - alert: WorkloadClusterAppFailed
       annotations:
         description: '{{`Workload Cluster App {{ $labels.name }}, version {{ $labels.version }} is {{if $labels.status }} in {{ $labels.status }} state. {{else}} not installed. {{end}}`}}'
         opsrecipe: app-failed/
-      expr: label_replace(app_operator_app_info{status!~"(?i:(deployed|cordoned))", catalog!~"control-plane-.*", name!="cluster-autoscaler"}, "cluster_id", "$1", "namespace", "(([^g]|g[^i]|gi[^a]|gia[^n]|gian[^t]|giant[^s]|giants[^w]|giantsw[^a]|giantswa[^r]|giantswar[^m])*)") == 1
+      expr: label_replace(app_operator_app_info{status!~"(?i:(deployed|cordoned))", catalog!~"control-plane-.*"}, "cluster_id", "$1", "namespace", "(([^g]|g[^i]|gi[^a]|gia[^n]|gian[^t]|giant[^s]|giants[^w]|giantsw[^a]|giantswa[^r]|giantswar[^m])*)") == 1
```
Comment (rossf7) [inline helm/prometheus-rules/templates/alerting-rules/app.rules.yml:129]: We need the replace magic to set the cluster ID. Current alerts show as MC apps

Screen Shot 2021-06-29 at 6 16 46 PM

**Comment by rossf7** [comment]
Tested on gauss.

File: helm/prometheus-rules/templates/alerting-rules/app.rules.yml (hunk 1 above)
Comment (tomahawk28) [reply helm/prometheus-rules/templates/alerting-rules/app.rules.yml:129]: To me, the only difference looks like coming from the deletion of `name!="cluster-autoscaler"}` is this right?

Comment (rossf7) [reply helm/prometheus-rules/templates/alerting-rules/app.rules.yml:129]: Yes, sorry I added the comment in the  wrong place.

File: helm/prometheus-rules/templates/alerting-rules/app.rules.yml [hunk 2]
```diff
@@ -124,6 +138,20 @@ spec:
         sig: none
         team: batman
         topic: releng
+    - alert: WorkloadClusterAppPendingUpdate
+      annotations:
+        description: 'Current version of {{`App {{ $labels.name }} is {{ $labels.deployed_version }} but it should be {{ $labels.version }}.`}}'
+      expr: label_replace(app_operator_app_info{catalog!~"control-plane-.*", deployed_version!="", status="deployed", version_mismatch="true"}, "cluster_id", "$1", "namespace", "(([^g]|g[^i]|gi[^a]|gia[^n]|gian[^t]|giant[^s]|giants[^w]|giantsw[^a]|giantswa[^r]|giantswar[^m])*)") == 1
```
Comment (rossf7) [inline helm/prometheus-rules/templates/alerting-rules/app.rules.yml:144]: This is where we need the replace so the workload cluster ID is shown in the alerts.

**Review by ubergesundheit — APPROVED** [review]

**Review by tomahawk28 — APPROVED** [review]

**Pull request merged**